### PR TITLE
Scalar types numpy

### LINKED ISSUE #571: Incorrect dtype with numpy floats
```python
import scipp as sp
import numpy as np

x = np.float32(0)
sp.Variable(x) 
```
gives
```
<scipp.Variable>          int64      [dimensionless]  ()  [0]
```

According the documentation python searches for the overload without conversion, the picks up the first for convertible type. As far as numpy types might be not the same as C++ types the overload for `double` or `float` is chosen dependently on the order of definition. @SimonHeybrock , any comments?

### PR DESCRIPTION
**Pull request by igudich**
fixes #571

### CODE REVIEW

**Comment by DanNixon** [comment]
The Numpy version in the Conda config will also need updated.

**Comment by igudich** [comment]
> The Numpy version in the Conda config will also need updated.

@DanNixon, @SimonHeybrock, @OwenArnold  could we somehow use one source of dependencies for python packages? Some script to generate conda config from requriments.txt or something like in here https://stackoverflow.com/questions/35245401/combining-conda-environment-yml-with-pip-requirements-txt?

**Comment by DanNixon** [comment]
> > The Numpy version in the Conda config will also need updated.
> 
> @DanNixon, @SimonHeybrock, @OwenArnold could we somehow use one source of dependencies for python packages? Some script to generate conda config from requriments.txt or something like in here [stackoverflow.com/questions/35245401/combining-conda-environment-yml-with-pip-requirements-txt](https://stackoverflow.com/questions/35245401/combining-conda-environment-yml-with-pip-requirements-txt)?

I'm not overly convinced generating one from the other is worth the effort, perhaps we just do away with `requirements.txt` and run the Python tests in the Conda environment.

**Comment by igudich** [comment]
> > > The Numpy version in the Conda config will also need updated.
> > 
> > 
> > @DanNixon, @SimonHeybrock, @OwenArnold could we somehow use one source of dependencies for python packages? Some script to generate conda config from requriments.txt or something like in here [stackoverflow.com/questions/35245401/combining-conda-environment-yml-with-pip-requirements-txt](https://stackoverflow.com/questions/35245401/combining-conda-environment-yml-with-pip-requirements-txt)?
> 
> I'm not overly convinced generating one from the other is worth the effort, perhaps we just do away with `requirements.txt` and run the Python tests in the Conda environment.

Anyhow, the goal is just to have dependencies in one place.

File: python/variable.cpp [hunk 1]
```diff
@@ -221,6 +242,12 @@ void init_variable(py::module &m) {
 
   py::implicitly_convertible<Variable, VariableConstProxy>();
 
+  m.def("make_variable_0d",
```
Comment (SimonHeybrock) [inline python/variable.cpp]: Can we "hide" this better? Maybe name `_make_variable_0d` or soemthing?

**Comment by DanNixon** [comment]
> Anyhow, the goal is just to have dependencies in one place.

Yes. That's what my suggestion achieves.

File: python/tests/test_variable.py [hunk 2]
```diff
@@ -551,11 +551,14 @@ def test_make_variable_from_unit_scalar_mult_div():
...
-    var = sp.Variable(v[sp.Dim.X, 0])
+    var = sp.Variable(np.float32())
     var.unit = sp.units.m
     assert var == np.float32(0.0) * sp.units.m
     var.unit = sp.units.m**(-1)
     assert var == np.float32(0.0) / sp.units.m
+
+
+def test_construct_0d_numpy():
+    v = sp.Variable([sp.Dim.X], values=np.array([0]), dtype=np.float32)
```
Comment (DanNixon) [inline python/tests/test_variable.py]: Is this test sufficient? Unless I misunderstood, arrays always worked as expected and the issue only arose when creating a `Variable` from a single Numpy scalar type.

Comment (igudich) [reply python/tests/test_variable.py]: > Is this test sufficient? Unless I misunderstood, arrays always worked as expected and the issue only arose when creating a `Variable` from a single Numpy scalar type.

Yes, exactly.

Comment (DanNixon) [reply python/tests/test_variable.py]: But that isn't what is being tested here, right? Should the test not be pretty much the code listed in the description of #571?

Comment (igudich) [reply python/tests/test_variable.py]: It is the same to lines below `assert var == sp.Variable(np.float32())`

Comment (DanNixon) [reply python/tests/test_variable.py]: Ah, yes I see now.
Is there a reason you don't just do, to me this makes what you are actually testing clearer:
```python
v = sc.Variable(np.float32())
assert v.dtype == sc.dtype.float
```

**Pull request merged**